en/docs/InstallAndSetup/UpgradingWSO2APIManager/upgrading-from-200-to-300.md

```diff
@@ -1,316 +1,2050 @@
+        driver = "org.postgresql.Driver"
+        validationQuery = "SELECT 1; COMMIT"
+        ```
 
-`/sequences/_build_.xml`
+        ```tab="Oracle"
+        [database.apim_db]
+        type = "oracle"
+        url = "jdbc:oracle:thin:@localhost:1521/mig_am_db"
+        username = "username"
+        password = "password"
+        driver = "oracle.jdbc.driver.OracleDriver"
+        validationQuery = "SELECT 1 FROM DUAL"
+        ```
 
-`/sequences/_cors_request_handler_.xml`
+        ```tab="DB2"
+        [database.apim_db]
+        type = "db2"
+        url = "jdbc:db2://localhost:50000/mig_am_db"
+        username = "username"
+        password = "password"
+        driver = "com.ibm.db2.jcc.DB2Driver"
+        validationQuery = "SELECT 1 FROM SYSIBM.SYSDUMMY1"
+        ```
+    
+    !!! note
+        It is recommended to use the default H2 database for the `WSO2_MB_STORE_DB` database in API-Manager. So do **not** migrate `WSO2_MB_STORE_DB` database from API-M 2.0.0 version to API-M 3.0.0 version, and use the **default H2** `WSO2_MB_STORE_DB` database available in API-M 3.0.0 version.
 
-`/sequences/fault.xml`
+4.  Update `<API-M_3.0.0_HOME>/repository/resources/conf/default.json` file by pointing to the WSO2UM_DB.
 
-`/sequences/main.xml`
+    ```
+    "realm_manager": {
+        "data_source": "WSO2USER_DB",
+        "properties": {
+        "isCascadeDeleteEnabled": true,
+        "initializeNewClaimManager": true
+        }
+    }
+    ```
 
-`/sequences/_production_key_error_.xml`
+5.  Copy the relevant JDBC driver to the `<API-M_3.0.0_HOME>/repository/components/lib` folder.
 
-`/sequences/_resource_mismatch_handler_.xml`
+    !!! info
+        In API-M 3.0.0, you do not need to configure the registry configurations as you did in the `<OLD_API-M_HOME>/repository/conf/registry.xml` file and the user database configurations as you did in in the `<OLD_API-M_HOME>/repository/conf/user-mgt.xml` file, as those configurations have been handled internally.
 
-`/sequences/_sandbox_key_error_.xml`
+6.  Move all your Synapse configurations to API-M 3.0.0 pack.
+    -   Move your Synapse super tenant configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/deployment/server/synapse-configs/default` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/deployment/server/synapse-configs/default` directory with the copied contents.
 
-`/sequences/_throttle_out_handler_.xml`
+    -   Move all your tenant Synapse configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/tenants` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/tenants` directory with the copied contents.
 
-`/sequences/_token_fault_.xml`
+    !!! warning
+        When moving the Synapse configurations, **do not replace** the following set of files as they contain some modificatiosn in API-M 3.0.0 version.
 
-`/proxy-services/WorkflowCallbackService.xml`
+        -   /api/\_RevokeAPI_.xml
```

Comment:
Do we need this back slash?

Reply:
Yes, it is added to escape the symbol **_**. Otherwise it will be displayed as `/api/RevokeAPI.xml`